### PR TITLE
Cleans up some code to do with hunter stealth and pouncing, stealth cd starts at exit

### PR DESCRIPTION
**Pull request by QualityVan**
There's code in place for pouncing someone while stealthed, but it never got reached due to the relevant destealth signal firing at pounce start instead of end. It's more or less nothing (1 slowdown 3 stagger) since hunter already inherits runner's 2 second paralyze regardless of stealth.

Also cleans up some duplicate code (attacking out of stealth with harm and disarm intent is line for line identical) and some code that wouldn't be reached (once you unstealth via pounce you can't sneak attack).

Nerfs hunter stealth by making the five second cooldown start when you leave stealth instead of only having a cooldown when you first enter it and when you deactivate it manually, which was weird. Thoughts on this welcome, given its use and plasma cost I'm not sure the cooldown matters at all.

:cl:
fix: Hunter pounce out of stealth now correctly gets its tiny sneak attack bonus
code: Misc. hunter stealth code cleanup
balance: Hunter stealth cooldown starts on leaving stealth.
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
> Nerfs hunter stealth by making the five second cooldown start when you leave stealth instead of only having a cooldown when you first enter it and when you deactivate it manually, which was weird. Thoughts on this welcome, given its use and plasma cost I'm not sure the cooldown matters at all.

check if this doenst break anything, like keeping getting less visible when toggling it off instantly, the cooldown might just be so you dont break the skill.

**Comment by QualityVan** [comment]
> check if this doenst break anything, like keeping getting less visible when toggling it off instantly, the cooldown might just be so you dont break the skill.

Did that, stealth hard sets your alpha layer and the callback to enable sneak attacks after stealthing checks to make sure you're still hidden.

**Comment by SplinterGP** [comment]
> 
> 
> > check if this doenst break anything, like keeping getting less visible when toggling it off instantly, the cooldown might just be so you dont break the skill.
> 
> Did that, stealth hard sets your alpha layer and the callback to enable sneak attacks after stealthing checks to make sure you're still hidden.

understood

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 1]
```diff
@@ -194,27 +186,6 @@
...
-		staggerslow_stacks *= 2
-		flavour = "deadly"
-
-	owner.visible_message("<span class='danger'>\The [owner] strikes [target] with [flavour] precision!</span>", \
-	"<span class='danger'>We strike [target] with [flavour] precision!</span>")
-	target.adjust_stagger(staggerslow_stacks)
-	target.add_slowdown(staggerslow_stacks)
-
-	cancel_stealth()
-
```
[the remark this answers is not in the document]
Comment (QualityVan) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:217]: Unless you want some specific effect for disarm strikes distinct from harm ones, it's not doing anything. And if someone decides they do want that later, it's very easy to copy/paste.

[the remark this answers is not in the document]
Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:217]: just leave it as is please, its intended

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**